client/components/helpCentre/diagnosticInformation/SubscriptionInformation.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+import type { MembersDataApiResponse } from '../../../../shared/productResponse';
+import { isProduct } from '../../../../shared/productResponse';
+import { allRecurringProductsDetailFetcher } from '../../../utilities/productUtils';
+
+export const SubscriptionInformation = () => {
+	const [memberData, setMemberData] = useState<
+		MembersDataApiResponse | 'failed' | 'loading'
+	>('loading');
+
+	useEffect(() => {
```

Comment:
there is actually a hook in the codebase called `useAsyncLoader` which does an async fetch and returns a loading state that you could use here. It does a lot of the same things and its usage looks fairly similar to what you have here